### PR TITLE
My HORROR PR

### PR DESCRIPTION
**Pull request by velnias75**
@hasufell **DON'T** merge that PR!

What the hell is going wrong here? I renamed the files and did my changes and **that** utter nonsense is the result.

This is the second try and the the result is the same, the files are totally wrong renamed.

:warning:  Mr *Linux Torvalds* if you care of your life **undo** that git, remove it out of history, delete it from all minds and **never** think about doing something similar again. Stick to what you are an expert for: **LINUX** and keep off hands from VCS for the rest of the existence of this universe!!!!

### CODE REVIEW

**Comment by hasufell** [comment]
don't change anything, don't do force-push

**Comment by velnias75** [comment]
I can't do anything. I'm only trying to understand what the **hell** git has done here...
I'm inspecting the side-by-side diff now

**Comment by hasufell** [comment]
start ```gitk --all``` and see the history

File: games-board/netmaumau/netmaumau-0.10-r2.ebuild [hunk 1]
```diff
@@ -18,10 +18,12 @@ KEYWORDS="~amd64 ~x86"
 IUSE=""
 
 RDEPEND="
-	dev-qt/qtcore:4
-	dev-qt/qtgui:4
-	dev-qt/qtsvg:4
-	games-server/netmaumau:0/4"
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.10-r2.ebuild]: Here nothing critical went wrong

File: games-board/netmaumau/netmaumau-0.10-r2.ebuild [hunk 2]
```diff
@@ -18,10 +18,12 @@ KEYWORDS="~amd64 ~x86"
...
+	dev-qt/qtcore:4[exceptions]
+	dev-qt/qtgui:4[exceptions]
+	dev-qt/qtsvg:4[exceptions]
+	games-server/netmaumau:0/4
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.10-r2.ebuild]: The above deleted 0.10 had slot 4, ok

File: games-board/netmaumau/netmaumau-0.11-r1.ebuild [hunk 3]
```diff
@@ -18,10 +18,12 @@ KEYWORDS="~amd64 ~x86"
...
 
 RDEPEND="
-	dev-qt/qtcore:4
-	dev-qt/qtgui:4
-	dev-qt/qtsvg:4
-	games-server/netmaumau:0/4"
+	dev-qt/qtcore:4[exceptions]
+	dev-qt/qtgui:4[exceptions]
+	dev-qt/qtsvg:4[exceptions]
+	games-server/netmaumau:0/5
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.11-r1.ebuild]: 11 (below) has slot 5, ok

File: games-board/netmaumau/netmaumau-0.13-r1.ebuild [hunk 4]
```diff
@@ -18,11 +18,13 @@ KEYWORDS="~amd64 ~x86"
 IUSE="espeak"
 
 RDEPEND="
-	dev-qt/qtcore:4
-	dev-qt/qtgui:4
-	dev-qt/qtsvg:4
-	espeak? ( app-accessibility/espeak )
-	games-server/netmaumau:0/6"
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.13-r1.ebuild]: ok

File: games-board/netmaumau/netmaumau-0.13-r1.ebuild [hunk 5]
```diff
@@ -18,11 +18,13 @@ KEYWORDS="~amd64 ~x86"
...
+	dev-qt/qtcore:4[exceptions]
+	dev-qt/qtgui:4[exceptions]
+	dev-qt/qtsvg:4[exceptions]
+	espeak? ( || ( app-accessibility/espeak[portaudio] app-accessibility/espeak[pulseaudio] ) )
+	games-server/netmaumau:0/6
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.13-r1.ebuild]: ok

**Comment by hasufell** [comment]
So what exactly is wrong about it?

File: games-board/netmaumau/netmaumau-0.4-r2.ebuild [hunk 6]
```diff
@@ -18,10 +18,12 @@ KEYWORDS="~amd64 ~x86"
 IUSE=""
 
 RDEPEND="
-	dev-qt/qtcore:4
-	dev-qt/qtgui:4
-	dev-qt/qtsvg:4
-	games-server/netmaumau:0/1"
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.4-r2.ebuild]: ok

File: games-board/netmaumau/netmaumau-0.4-r2.ebuild [hunk 7]
```diff
@@ -18,10 +18,12 @@ KEYWORDS="~amd64 ~x86"
...
+	dev-qt/qtcore:4[exceptions]
+	dev-qt/qtgui:4[exceptions]
+	dev-qt/qtsvg:4[exceptions]
+	games-server/netmaumau:0/1
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.4-r2.ebuild]: ok

**Comment by velnias75** [comment]
The left file name doesn't correspond to the right, so I need to compare the 
slots now.

If they are ok, we can merge. But what the hell went wrong only Linux can 
know.

This is far from being intuitive...

**Comment by hasufell** [comment]
The git HISTORY looks ok... you are just ONE commit ahead of my master branch.

It may just be that git tries to make sense out of the file-moves... it compares them and if they match e.g. 80% or more it considers it a file-move instead of a two-step remove-then-add thing. When doing lots of renames it may just be that it couldn't figure out what corresponds to what... but that doesn't matter for the END RESULT. It should be the same.

File: games-board/netmaumau/netmaumau-0.6-r3.ebuild [hunk 8]
```diff
@@ -18,10 +18,12 @@ KEYWORDS="~amd64 ~x86"
 IUSE=""
 
 RDEPEND="
-	dev-qt/qtcore:4
-	dev-qt/qtgui:4
-	dev-qt/qtsvg:4
-	games-server/netmaumau:0/2"
```
Comment (velnias75) [inline games-board/netmaumau/netmaumau-0.6-r3.ebuild]: 0.7/0.6 -> 0.6/07 cross but ok

**Comment by hasufell** [comment]
https://stackoverflow.com/questions/15031576/how-can-i-prevent-git-from-thinking-i-did-a-rename

**Comment by velnias75** [comment]
Ok, we give it a chance and merge? I'll check it afterwards (tomorrow morning) and will fix it if neccessary.

The commit difference is for 0.14

**Comment by velnias75** [comment]
BTW: thanks for the reminder, I have to accept an answer at stackoverflow :grinning:

**Comment by hasufell** [comment]
compare what github shows you with ```git show```... the latter does it more verbose

**Comment by hasufell** [comment]
One way around that is... bump first... remove old stuff in a SECOND commit.

**Comment by velnias75** [comment]
I checked it, no file is lost and it seems the result will stay the same. We can give it a try.
If people are so crazy to persist on older versions of the client, okay, they have bad luck until tomorrow if something went wrong. But again git gives **absolutely** unneccessary work...

**Comment by velnias75** [comment]
`equery g games-board/netmaumau` revealed all is okay, too. The slot dependencies are the same as before.

**Comment by velnias75** [comment]
Ok, a final test I'll do tomorrow but I don't expect any changes.
For now I go to bed. My :cat: is already waiting for me :smile_cat:

**Comment by velnias75** [comment]
:information_source: all tests have succeeded :smile_cat:

**Pull request merged**